jbmc/src/java_bytecode/README.md

```diff
@@ -35,9 +35,108 @@ To be documented.
 
 To be documented.
 
-\section java-bytecode-remove-java-new Remove java new
+\section java-bytecode-remove-java-new Remove `new`, `newarray` and `multianewarray` bytecode operators
 
-To be documented.
+\ref remove_java_new.h is responsible for converting the `new`, `newarray` and
+`multianewarray` Java bytecode operation into \ref codet. Specifically it
+converts the bytecode instruction into:  - An ALLOCATE with the size of the
+object being created  - An assignment to the value zeroing its contents  - If an
+array, initializing the size and data components  - If a multi-dimensional
+array, recursively calling `java_new` on each sub array
+
+An ALLOCATE is a \ref side_effect_exprt that is interpreted by \ref
+goto_symext::symex_allocate
+
+_Note: it does not call the constructor as this is done by a separate
+java_bytecode operation._
+
+\subsection java_objects Java Objects (`new`)
+
+The basic `new` operation is represented in Java bytecode by the `new` op
+
+These are converted by \ref remove_java_newt::lower_java_new
+
+For example, the following Java code:
+
+```java
+TestClass f = new TestClass();
+```
+
+Which is represented as the following Java bytecode:
+
+```
+0: new           #2                  // class TestClass
+3: dup
+4: invokespecial #3                  // Method "<init>":()V
+```
+
+The first instruction only is translated into the following `codet`s:
+
+```cpp
+tmp_object1 = side_effect_exprt(ALLOCATE, sizeof(TestClass))
+*tmp_object1 = struct_exprt{.component = 0, ... }
+```
+
+For more details about the zero expression see \ref expr_initializert
+
+\subsection oned_arrays Single Dimensional Arrays (`newarray`)
+
+The new Java array operation is represented in Java bytecode by the `newarray`
+operation.
+
+These are converted by \ref remove_java_newt::lower_java_new_array
+
+See TODO: DOC-20: for details on how arrays are represented in codet. It first
```

Comment:
This reference is not useful because DOC-20 is nothing public. Refer to some entity within this document or create a new ticket that covers this TODO.